### PR TITLE
Add generic

### PR DESCRIPTION
**Pull request by oscarbc96**
- Added `Principal`property.
- Modified `Statement` property to work with `Principal` property
- Added `Generic`property. Any property under this class will be casted to an existing model of pycfmodel if possible.
- Modified `GenericResource`, `ESDomainProperties`, `OpenSearchDomainProperties` and `S3BucketProperties` to work with `Generic` property
- Update `CLOUDFORMATION_ACTIONS`

### CODE REVIEW

File: pycfmodel/model/resources/properties/statement.py [hunk 1]
```diff
@@ -109,11 +117,11 @@ def get_principal_list(self) -> List[ResolvableStr]:
                 principal_list.append(principals)
             elif is_resolvable_dict(principals):
                 principal_list.append(principals)
-            elif isinstance(principals, dict):
-                for value in principals.values():
-                    if isinstance(value, (str, Dict)):
+            elif isinstance(principals, Principal):
+                for value in principals.dict().values():
+                    if isinstance(value, (str, dict)):
```
Comment (oscarbc96) [inline pycfmodel/model/resources/properties/statement.py:122]: add test

File: pycfmodel/model/resources/properties/statement.py [hunk 2]
```diff
@@ -11,7 +11,15 @@
 
 logger = logging.getLogger(__file__)
 
-PrincipalTypes = Union[ResolvableStr, List[ResolvableStr], Dict[str, Union[ResolvableStr, List[ResolvableStr]]]]
+
+class Principal(Property):
```
Comment (oscarbc96) [inline pycfmodel/model/resources/properties/statement.py:15]: add test

File: pycfmodel/model/generic.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                auxiliar_list.append(v)
+            value = auxiliar_list
+
+        if isinstance(value, dict):
+            value = Generic.parse_obj(value)
+
+        return value
+
+
+class Generic(BaseModel):
```
Comment (oscarbc96) [inline pycfmodel/model/generic.py:57]: add test

**Review by ocrawford555 — COMMENTED** [review]
This is awesome work Oscar! Thanks for adding the additional test around handling a generic resource with a `PolicyDocument`. Left some small comments for review (mostly documentation fixes).

File: pycfmodel/model/types.py [hunk 4]
```diff
@@ -47,8 +48,7 @@ def validate(cls, value: NetworkType) -> IPv6Network:
 ResolvableInt = Resolvable[int]
 ResolvableDate = Resolvable[date]
 ResolvableDatetime = Resolvable[datetime]
-ResolvableBool = Resolvable[bool]
-ResolvableDict = Resolvable[dict]
+ResolvableBool = Resolvable[StrictBool]
```
[the remark this answers is not in the document]
Comment (ignaciobolonio) [reply pycfmodel/model/types.py:51]: I think it is to avoid strings such as `1` to be parsed as booleans

[the remark this answers is not in the document]
Comment (oscarbc96) [reply pycfmodel/model/types.py:51]: Plus one on @ignaciobolonio comment

**Review by ignaciobolonio — APPROVED** [review]
Amazing work! The part for `Generic` and `_Auxiliar` classes is top code 🙌

**Review by ocrawford555 — APPROVED** [review]

**Pull request merged**